specs/eventing/control-plane.md

```diff
@@ -0,0 +1,1097 @@
+before the data plane programming was complete and the Trigger was updated to
+set the `Ready` condition to `true`.)
+
+If multiple Triggers match an event, one event delivery MUST be generated for
+each match; duplicate matches with the same destination MUST each generate a
+separate event delivery attempts, one per Trigger match. The implementation MAY
+attach additional event attributes or other metaata distinguishing between these
+deliveries. The implementation MUST NOT modify the event payload in this
+process.
+
+Reply events generated during event delivery MUST be re-enqueued by the Broker
+in the same way as events delivered to the Broker's Addressable URL. Reply
+events re-enqued in this maanner MUST be evaluated against all triggers
+associated with the Broker, including the Trigger that generated the reply.
+Implementations MAY implement event-loop detection; it is RECOMMENDED that any
+such controls be documented to end-users. Implementations MAY avoid using HTTP
+to deliver event replies to the Broker's event-delivery input and instead use an
+internal queueing mechanism.
+
+## Topology Based Routing
+
+A Channel MUST publish a URL at `status.address.url` when it is able to receive
+events. This URL MUST accept CloudEvents in both the
+[Binary Content Mode](https://github.com/cloudevents/spec/blob/v1.0.1/http-protocol-binding.md#31-binary-content-mode)
+and
+[Structured Content Mode](https://github.com/cloudevents/spec/blob/v1.0.1/http-protocol-binding.md#32-structured-content-mode)
+HTTP formats. Before sending an HTTP response, the Channel MUST durably enqueue
+the event (be able to deliver with retry without receiving the event again).
+
+For each event received by the Channel, the Channel MUST deliver the event to
+each associated Subscription **once** (where "associated" means a Subscription
+with a `spec.channel` which references the Channel). If the Subscription has a
+`Ready` condition of `true` when the event is evaluated, the Channel MUST
+forward the event as described in
+[event delivery as described below](#event-delivery). The Channel MAY also
+forward events to associated Subscriptions for with the `Ready` condition is not
+currently `true`. (One example: a Subscription which is in the process of being
+programmed in the Channel data plane might receive _some_ events before the data
+plane programming was complete and the Subscription was updated to set the
+`Ready` condition to `true`.)
+
+If multiple Subscriptions with the same destination are associated with the same
+Channel, each Subscription MUST generate one delivery attempt per Subscription.
+The implementation MAY attach additional event attributes or other metadata
+distinguishing between these deliveries. The implementation MUST NOT modify the
+event payload in this process.
+
+## Event Delivery
+
+Once a Trigger or Subscription has decided to deliver an event, it MUST do the
+following:
+
+1. Resolve all URLs and delivery options.
```

Comment:
I'd shy away from the MUST here only because the URL resolution doesn't have to happen for each event?

Reply:
I meant that they MUST be resolved, but that could just mean copying from `status`. I'll clarify.